pkg/querier/batch/stream.go

```diff
@@ -46,52 +62,67 @@ func mergeStreams(left, right batchStream, result batchStream, size int) batchSt
 	resultLen := 1 // Number of batches in the final result.
 	b := &result[0]
 
-	// This function adds a new batch to the result
-	// if the current batch being appended is full.
-	checkForFullBatch := func() {
-		if b.Index == size {
-			// The batch reached it intended size.
-			// Add another batch the result
-			// and use it for further appending.
-
-			// The Index is the place at which new sample
-			// has to be appended, hence it tells the length.
-			b.Length = b.Index
-			resultLen++
-			if resultLen > len(result) {
-				// It is possible that result can grow longer
-				// then the one provided.
-				result = append(result, promchunk.Batch{})
-			}
-			b = &result[resultLen-1]
+	// This function adds a new batch to the result.
+	nextBatch := func(valueType chunkenc.ValueType) {
+		// The Index is the place at which new sample
+		// has to be appended, hence it tells the length.
+		b.Length = b.Index
+		resultLen++
+		if resultLen > len(result) {
+			// It is possible that result can grow longer
+			// then the one provided.
+			result = append(result, promchunk.Batch{})
 		}
+		b = &result[resultLen-1]
+		b.ValType = valueType
 	}
 
-	for left.hasNext() && right.hasNext() {
-		checkForFullBatch()
+	populateVal := func(bs batchStream, valueType chunkenc.ValueType) {
+		if b.Index == 0 {
+			b.ValType = valueType
+		} else if b.Index == size || b.ValType != valueType {
+			// The batch reached it intended size or a new value type is used.
+			// Add another batch to the result and use it for further appending.
```

Comment:
Is it possible to change the value type?

Reply:
Each batch will only have one value type.